socs/agents/starcam/agent.py

```diff
@@ -0,0 +1,197 @@
+                return False, "Could not acquire lock"
+            self.log.info("Sending commands")
+            self.starcam_Helper.pack_cmds()
+            self.starcam_Helper.send_cmds()
+        return True, "Sent commands to starcamera"
+
+    @ocs_agent.param('_')
+    def acq(self, session, params=None):
+        """start_acq(test_mode=False)
+        **Process** - Acquire data from starcam and write to feed.
+
+        Parameters:
+            test_mode (bool, optional): Run the acq process loop only once. This is meant only for testing. Default is false.
+        """
+        if params is None:
+            params = {}
+        with self.lock.acquire_timeout(timeout=100, job='init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock"
+            session.set_status('running')
+            self.log.info("Starting acquisition")
+            self.take_data = True
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'astrometry',
+                    'data': {}
+                }
+                c_time_reading, gmt_reading, blob_num_reading, obs_ra_reading, astrom_ra_reading, obs_dec_reading, astrom_dec_reading, fr_reading, ps_reading, alt_reading, az_reading, ir_reading, astrom_solve_time_reading, camera_time_reading = self.starcam_Helper.get_astrom_data()
+                data['data']['c_time'] = c_time_reading
+                data['data']['gmt'] = gmt_reading
+                data['data']['blob_num'] = blob_num_reading
+                data['data']['obs_ra'] = obs_ra_reading
+                data['data']['astrom_ra'] = astrom_ra_reading
+                data['data']['obs_dec'] = obs_dec_reading
+                data['data']['astrom_dec'] = astrom_dec_reading
+                data['data']['fr'] = fr_reading
+                data['data']['ps'] = ps_reading
+                data['data']['alt'] = alt_reading
+                data['data']['az'] = az_reading
+                data['data']['ir'] = ir_reading
+                data['data']['astrom_solve_time'] = astrom_solve_time_reading
+                data['data']['camera_time'] = camera_time_reading
+                session.data.update(data['data'])
+                self.agent.publish_to_feed('starcamera', data)
+        return True, 'Acquisition exited cleanly'
+
+    def _stop_acq(self, session, params):
+        ok = False
+        if self.take_data:
+            session.set_status('stopping')
+            self.take_data = False
+            ok = True
+            # self.starcam_Helper.close()
+        return (ok, {True: 'Requested process to stop', False: 'Failed to request process stop.'}[ok])
+
+
+def add_agent_args(parser_in=None):
+    if parser_in is None:
+        from argparse import ArgumentParser as A
+        parser_in = A()
```

Comment:
Line 178 is actually still needed, `parser_in` is used below.

But along these lines, move the argparse import to the top of the file as just `import argparse` and make these lines:
```
    if parser is None:
        parser = argparse.ArgumentParser()
```
for consistency with other agents. (Also make the argument `parser` instead of `parser_in`.)